File: service/router/panel/networkSetting.go
package panel

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitNetworkSetting(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiPanel.NetworkSetting
	r := router.Group("", middleware.LoginInterceptor, middleware.AdminInterceptor)
	{
		r.POST("/panel/networkSetting/getFavicon", api.GetFavicon)
		r.POST("/panel/networkSetting/setFavicon", api.SetFavicon)
		r.POST("/panel/networkSetting/testFavicon", api.TestFavicon)
	}
}

File: src/api/panel/networkSetting.ts
import { post } from '@/utils/request'

export function getFaviconNetwork<T>() {
  return post<T>({
    url: '/panel/networkSetting/getFavicon',
  })
}

export function setFaviconNetwork<T>(data: NetworkSetting.FaviconNetworkSetting) {
  return post<T>({
    url: '/panel/networkSetting/setFavicon',
    data,
  })
}

export function testFaviconNetwork<T>(data: NetworkSetting.FaviconNetworkTestRequest) {
  return post<T>({
    url: '/panel/networkSetting/testFavicon',
    data,
  })
}
